pymeasure/instruments/keithley/keithley6487.py

```diff
@@ -0,0 +1,395 @@
+        """
+    )
+
+    current_range = Instrument.control(
+        ":SENS:CURR:RANG?", ":SENS:CURR:RANG:AUTO 0; :SEND:CURR:RANG %f",
+        """A floating point property that controls the measurement the current
+        range in Amps, which can take values between -0.021 and +0.021 A.
+        Auto-range is disabled when this property is set""",
+        validator=truncated_range,
+        values=[-0.021, 0.021]
+    )
+
+    current_nplc = Instrument.control(
+        ":SENS:CURR:NPLC?", ":SENS:CURR:NPLC %f",
+        """ A floating point property that controls the number of power line cycles
+        (NPLC) for the DC current measurements, which sets the integration period
+        and measurement speed. Takes values from 0.001 to 6.0 or 5.0, where 6.0
+        is 60Hz and 5.0 is 50Hz"""
+    )
+
+    ###########
+    # Trigger #
+    ###########
+
+    trigger_count = Instrument.control(
+        ":TRIG:COUN?", ":TRIG:COUN %d",
+        """ An integer property that controls the trigger count,
+        which can take values from 1 to 2048. """,
+        validator=truncated_range,
+        values=[1, 2048],
+        cast=int
+    )
+    trigger_count_inf = Instrument.control(
+        ":TRIG:COUNT?", "TRIG:COUNT %s",
+        """A string that controls the trigger count,
+        sets trigger count to INF mode instead of an integer""",
+        validator=strict_discrete_set,
+        values=['INF'],
+        cast=str
+    )
+
+    trigger_delay = Instrument.control(
+        ":TRIG:SEQ:DEL?", ":TRIG:SEQ:DEL %f",
+        """ A floating point property that controls the trigger delay
+        in seconds, which can take values from 0 to 999.9999 s. """,
+        validator=truncated_range,
+        values=[0, 999.9999]
+    )
+
+    ###########
+    # Filters #
+    ###########
+
+    filter_type = Instrument.control(
+        ":SENS:AVER:TCON?", ":SENS:AVER:TCON %s",
+        """ A String property that controls the filter's type.
+        REP : Repeating filter
+        MOV : Moving filter""",
+        validator=strict_discrete_set,
+        values=['REP', 'MOV'],
+        map_values=False)
+
+    filter_count = Instrument.control(
+        ":SENS:AVER:COUNT?", ":SENS:AVER:COUNT %d",
+        """ A integer property that controls the number of readings that are
+        acquired and stored in the filter buffer for the averaging""",
+        validator=truncated_range,
+        values=[2, 100],
+        cast=int)
+
+    filter_state = Instrument.control(
+        ":SENS:AVER?", ":SENS:AVER %s",
+        """ A string property that controls if the filter is active.""",
+        validator=strict_discrete_set,
+        values=['ON', 'OFF'],
+        map_values=False)
+
+    ####################
+    #   Methods        #
+    ####################
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter, "Keithley 6487 Picoammeter", **kwargs
+        )
+
+    def enable_source(self):
```

Comment:
I think these methods `enable_source/disable_source` are not needed, since you have already a property defined to do this.
PyMeasure encourages the usage of properties whenever it's possible.